src/usb/uf2/ghostfat.c

```diff
@@ -63,7 +63,29 @@ struct TextFile {
 //
 //--------------------------------------------------------------------+
 
-#define NUM_FAT_BLOCKS CFG_UF2_NUM_BLOCKS
+#define BPB_SECTOR_SIZE           ( 512)
+#define BPB_SECTORS_PER_CLUSTER   (   1)
+#define BPB_RESERVED_SECTORS      (   1)
+#define BPB_NUMBER_OF_FATS        (   2)
+#define BPB_ROOT_DIR_ENTRIES      (  64)
+#define BPB_TOTAL_SECTORS         CFG_UF2_NUM_BLOCKS
+#define BPB_MEDIA_DESCRIPTOR_BYTE (0xF8)
+#define FAT_ENTRY_SIZE            (2)
+#define FAT_ENTRIES_PER_SECTOR    (BPB_SECTOR_SIZE / FAT_ENTRY_SIZE)
+// NOTE: MS specification explicitly allows FAT to be larger than necessary
+#define BPB_SECTORS_PER_FAT       ( (BPB_TOTAL_SECTORS / FAT_ENTRIES_PER_SECTOR) + \
+                                   ((BPB_TOTAL_SECTORS % FAT_ENTRIES_PER_SECTOR) ? 1 : 0))
+#define DIRENTRIES_PER_SECTOR     (BPB_SECTOR_SIZE/sizeof(DirEntry))
+#define ROOT_DIR_SECTOR_COUNT     (BPB_ROOT_DIR_ENTRIES/DIRENTRIES_PER_SECTOR)
+
+STATIC_ASSERT(BPB_SECTOR_SIZE                              ==       512); // GhostFAT does not support other sector sizes (currently)
+STATIC_ASSERT(BPB_SECTORS_PER_CLUSTER                      ==         1); // GhostFAT presumes one sector == one cluster (for simplicity)
+STATIC_ASSERT(BPB_NUMBER_OF_FATS                           ==         2); // FAT highest compatibility
+STATIC_ASSERT(sizeof(DirEntry)                             ==        32); // FAT requirement
```

Comment:
this is already checked after the DirEntry definition.. Look at this now, I think we don't really need to perform these static assert, just put an warning before these BPB and tell user not to change it. Basically we static check thing that could be wrong like the `__attribute__((packed)) DirEntry;` not the actual macro with fixed magic number

Reply:
static assertions catch user error.
user error can include accidental and unintentional changes (as in your example).

user ignorance can also cause a user error.
a later editor (not you) might change something without full understanding.

therefore, static assert also is useful for making explicit code assumptions.

This code had many assumptions.

Reply:
we don't need to have static check on every magic value, those magic constant is part of any C source.